### ISSUE TITLE
Sitename with ampersand causes `drush up backdrop` failure

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/drush#91]
It gives a success message but the files are not replaced correctly if the sitename contains an asterisk. 

I suppose we could pretty easily add another `str_replace` [on this line](https://github.com/backdrop-contrib/drush/blob/d16a9d782c78b659abd117a04ae45355e37e7125/commands/pm/backdrop_pm.drush.inc#L310), but there are probably other problem characters out there. We may need to figure a better way to handle this part. Does [Transliteration](https://backdropcms.org/project/transliteration) provide a function that may be useful here?


```
$ drush up backdrop

                Current Backdrop Version: 1.6.0
                        Proposed Backdrop Version: 1.6.3

        Code updates will be made to Backdrop core.

        WARNING: Updating core will discard any
        modifications made to Backdrop core files. If you have made any
        modifications to these files, please back them up before updating so that
        you can re-create your modifications in the updated version of the file.

        Note: Updating core can potentially break your site.
        It is NOT recommended to update production sites without prior testing.

        Do you wish to continue? (y or n): y
        Backing up current core folder to /home/username/.drush/name_&_name2-core-1.6.0
sh: _name2-core-1.6.0: command not found
mv: cannot move `/home/username/public_html/core' to `/home/username/.drush/name_/core': File exists
No database updates required                                                                                                                                                                                                                                    [success]
Finished performing updates.                                                                                                                                                                                                                                    [ok]

        You can retrieve your old core files, should you need
          to from: /home/username/.drush/core-1.6.0
        Success: Backdrop updated to 1.6.3.
```

**Comment by serundeputy** [comment]
@laryn Thanks for reporting.  I'll find a more robust string sanitization method!

**Comment by laryn** [comment]
@serundeputy Can you use this?

https://github.com/backdrop/backdrop/blob/1.x/core/includes/transliteration.inc#L26

**Comment by serundeputy** [comment]
@laryn I think this: https://github.com/backdrop/backdrop/blob/1.x/core/includes/transliteration.inc#L241

**Comment by serundeputy** [comment]
@laryn up for testing this out?
`PR`: https://github.com/backdrop-contrib/drush/pull/92

**Comment by laryn** [comment]
@serundeputy Sure, probably tomorrow.

**Comment by laryn** [comment]
I just tested on localhost without the new patch and it didn't give an error (allowed the ampersand in the filename). I'll try to spin up an outdated Backdrop on the server that gave the problem earlier and test there.

**Comment by laryn** [comment]
@serundeputy Tested again on the server that had the problem initially and it went flawlessly. (Initial test without the patch failed, applied the patch and had no problem).

**Comment by serundeputy** [comment]
Thanks for the report, research and testing @laryn !

fixed in a9b9138ef9dd3508d32701ac40e08c600d33ba23
